src/parser.js

```diff
@@ -734,20 +734,86 @@ parser.getFileContents = (originalFilename, partialPath) => {
     return { filedata, raw };
 };
 
+const normalizePath = (str) => path.join('', str);
+parser.getSourcesFileContents = (sources, entryFile) => {
+    const included = {};
+    const parentPathOf = (str) => (m = /(.*)\/(?!\/)(.*)\.huff/g.exec(str)) && m[1];
+    const recurse = (filename, parentPath) => {
+        const fileName = normalizePath(filename)
+        const relativePath = path.join(parentPath || '', fileName);
+        let fileString = sources[relativePath];
+        if (!fileString) {
+            if (!fs.existsSync(relativePath)) {
+                throw new Error(`Import not found: ${fileName} ${(parentPath && parentPath !== '.') ? `in ${parentPath}` : ''}`)
+            }
+            fileString = fs.readFileSync(relativePath, 'utf8');
+            console.log(`Warning: File ${relativePath} does not exist in sources, found in filesystem.`)
+        }
+        let formatted = parser.removeComments(fileString);
+        let imported = [];
+        let test = formatted.match(grammar.topLevel.IMPORT);
+        while (test !== null) {
+            const importStatement = formatted.match(grammar.topLevel.IMPORT);
+            const empty = ' '.repeat(importStatement[0].length);
+            formatted = empty + formatted.slice(importStatement[0].length);
```

Comment:
If you make the `recurse` function return the parsed file data, with import statements replaced with the data of the underlying file they refer to, you can replace `empty` with the return of `recurse(importedFilename, parentPathOf(relativePath))`, getting rid of quite a few `...` and a `reduce`.

i.e. have your recurse aggregate all imports into one source string, and return that.